core/src/consensus/tendermint/worker.rs

```diff
@@ -667,15 +673,10 @@ impl Worker {
                 self.request_messages_to_all(vote_step, &BitSet::all_set() - &self.votes_received);
                 if !self.already_generated_message() {
                     let block_hash = match &self.last_two_thirds_majority {
-                        TwoThirdsMajority::Empty => None,
-                        TwoThirdsMajority::Unlock(_) => None,
-                        TwoThirdsMajority::Lock(locked_view, block_hash) => {
-                            if locked_view == &self.view {
-                                Some(*block_hash)
-                            } else {
-                                None
-                            }
+                        TwoThirdsMajority::Lock(locked_view, block_hash) if locked_view == &self.view => {
```

Comment:
It seems that it's the same with the previous code. Is it?

Reply:
Yes, It's the pre-commit step. Do I need to separate it to another commit or just revert it ?

Reply:
Why do you change if it doesn't change the behavior?

Reply:
At first I was changing the pre-commit step and it helps to introduce checking block generation time and rejecting if irrelevant. After I had reverted it, I thought it still helps because it simplifies the routine. But it's meaningless in the sense that it doesn't change the behavior.

Reply:
And it makes code fragile when changing because you used the underscore arm.
When someone adds a new enumerator, the compiler knows where to fix it in the original code, but it doesn't know in this code.

Reply:
Ah, I understand. I couldn't think about that.